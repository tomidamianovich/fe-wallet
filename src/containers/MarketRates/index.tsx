import React from 'react';
import './styles/index.scss';
import { withTranslation, WithTranslation } from 'react-i18next';
import MaintenancePage from "../../components/MaintenancePage";

type Props = {} & WithTranslation;

const MarketRates: React.FC<Props> = ({t}) => 
  <div className="fe-wallet__market-rates">
    <MaintenancePage pageName={t('pages.market_rates')} />
  </div>
;

export default withTranslation()(MarketRates);