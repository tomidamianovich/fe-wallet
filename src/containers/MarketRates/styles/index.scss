.fe-wallet__market-rates {
  display: flex;
  width: 100%;
}
