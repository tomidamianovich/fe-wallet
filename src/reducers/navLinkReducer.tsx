import * as actionTypes from '../actionTypes/NavLinkType'
import { NavLinkAction, NavLinkState } from '../utils/type'
import { NAV_LINKS } from '../utils/constants'

const initialState: NavLinkState = NAV_LINKS

const reducer = (
  state: NavLinkState = initialState,
  action: NavLinkAction,
): NavLinkState => {
  switch (action.type) {
    case actionTypes.SET_NAV_LINKS:
      return action.payload
  }
  return state
}

export default reducer
