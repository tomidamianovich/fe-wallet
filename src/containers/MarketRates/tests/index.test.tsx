import { render } from '@testing-library/react';
import MarketRates from '../';
import navLinkReducer from "../../../reducers/navLinkReducer"
import { Provider } from 'react-redux';
import { createStore } from "redux";

const store = createStore(navLinkReducer);

jest.mock('react-i18next', () => ({
  withTranslation: () => (Component: any) => {
    Component.defaultProps = {...Component.defaultProps, t: (translation: string) => translation};
    return Component;
  }
}));


test('renders Market Rates main page', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <MarketRates />
    </Provider>);
  expect(asFragment()).toMatchSnapshot();
});