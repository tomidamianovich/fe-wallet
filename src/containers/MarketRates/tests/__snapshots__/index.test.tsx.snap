// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Market Rates main page 1`] = `
<DocumentFragment>
  <div
    class="fe-wallet__market-rates"
  >
    <div
      class="fe-wallet__maintenance-page"
    >
      <h1>
        maintenance.title
      </h1>
      <h3>
        maintenance.subtitle.beginningmaintenance.subtitle.ending
      </h3>
    </div>
  </div>
</DocumentFragment>
`;
